pyartcd/pyartcd/exceptions.py

```diff
@@ -1,2 +1,43 @@
 class VerificationError(ValueError):
     pass
+
+
+class GithubApiExceptions(Exception):
+    """The super class for all GiHub API exceptions"""
+    pass
+
+
+class BranchCreationError(GithubApiExceptions):
+    """Exception raised when branch could not be created using GitHub API reuqest. Status code other than 201 is received"""
+
+
+class BaseBranchNotFound(GithubApiExceptions):
+    """Exception raised when base branch could not be found from the repo using the GitHub API"""
+
+
+class YmlFileKeyError(GithubApiExceptions):
+    """Exception raised when yml file obtained by parsing the response does not have the key we expect"""
+
+
+class JsonKeyError(GithubApiExceptions):
+    """Exception raised when JSON file obtained by parsing the response does not have the key we expect"""
+
+
+class PushError(GithubApiExceptions):
+    """Exception raised when trying to push to remote repo does not succeed"""
+
+
+class PrCreationError(GithubApiExceptions):
+    """Exception raised when PR could not be created by the GitHub API"""
+
+
+class BranchDeletionError(GithubApiExceptions):
+    """Exception raised if a branch could not be deleted using the GitHub API"""
+
+
+class BranchExists(GithubApiExceptions):
+    """Exception raised when branch already exists in the repo"""
+
+
+class UpstreamBranchSyncError(GithubApiExceptions):
+    """Exception raised when the branch cannot be synced with  upstream"""
```

Comment:
Actually you don't need so many exception types unless you catch them and do different error handlings.
A single GithubApiException with descriptive message is good enough.

Reply:
+1

Reply:
Done!